src/test/java/com/eighthlight/fabrial/test/server/TcpServerEchoIntegrationTest.java

```diff
@@ -0,0 +1,47 @@
+package com.eighthlight.fabrial.test.server;
+
+import com.eighthlight.fabrial.test.client.EchoRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class TcpServerEchoIntegrationTest extends TcpServerIntegrationTest {
+  @BeforeEach
+  void startAndConnect() throws IOException {
+    server.start();
+    client.connect();
+  }
+
+  @Test
+  void testEchoingString() throws IOException, ClassNotFoundException  {
```

Comment:
Where does the `ClassNotFoundException` in all these tests come from? You're not doing any class loading at runtime are you? Ahh, digging further, I see why in `EchoRequest.read()`, will comment there.